File: OtherLanguages/PHP/CrashReporter/Data/Crash.php
<?
class Crash
{
	public $id;
	public $applicationGuid;
	public $exception;

	public function read($reader)
	{
		$this->id = $reader->hasField(0) ? $reader->readInteger(0) : null;
		$this->applicationGuid = $reader->readString(1);
		$this->exception = $reader->readString(2);
	}

	public function write($writer)
	{
		if ($this->id!=null) $writer->writeValue(0, $this->id);
		$writer->writeString(1, $this->applicationGuid);
		$writer->writeString(2, $this->exception);
	}
}
?>

File: Sitana.Framework/CrashReporter/Data/Crash.cs
using System;
using Ebatianos.DataTransfer;

namespace Ebatianos.CrashReporter.Data
{
	public class Crash: IEcsStructure
	{
		public int? Id;
		public string ApplicationGuid;
		public string Exception;

        void IEcsStructure.Read(EcsReader reader)
		{
			Id = reader.HasField(0) ? (int?)reader.ReadInt32(0) : null;
			ApplicationGuid = reader.ReadString(1);
			Exception = reader.ReadString(2);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			if (Id!=null) writer.Write(0, Id.Value);
			writer.Write(1, ApplicationGuid);
			writer.Write(2, Exception);
		}
	}
}